File: demo-app/android/observe-sdk/src/main/java/com/observe/sdk/core/ObserveConfig.kt
package com.observe.sdk.core

/**
 * Configuration for Observe SDK
 * 
 * @param enabled Enable/disable SDK
 * @param autoStart Automatically start observation on init
 * @param eventBufferSize Number of events to buffer before export
 * @param maxStoredFiles Maximum number of event files to keep
 * @param exportEndpoint Optional network endpoint for event upload
 * @param debugMode Enable verbose logging
 * @param exportCryptoKeys Export TLS/SSL keys for traffic decryption (SECURITY WARNING: test builds only!)
 * @param bypassCertPinning Disable SSL certificate pinning (SECURITY WARNING: test builds only!)
 */
data class ObserveConfig(
    val enabled: Boolean = true,
    val autoStart: Boolean = true,
    val eventBufferSize: Int = 50,
    val maxStoredFiles: Int = 10,
    val exportEndpoint: String? = null,
    val debugMode: Boolean = true,
    val exportCryptoKeys: Boolean = false,
    val bypassCertPinning: Boolean = false
) {
    companion object {
        /**
         * Default configuration
         */
        fun default() = ObserveConfig()
        
        /**
         * Disabled configuration (for production)
         */
        fun disabled() = ObserveConfig(enabled = false)
        
        /**
         * Full observation mode with crypto key export
         * 
         * SECURITY WARNING:
         * This mode disables SSL security and exports encryption keys!
         * ONLY use in observe/test builds!
         */
        fun fullObservation() = ObserveConfig(
            enabled = true,
            autoStart = true,
            exportCryptoKeys = true,
            bypassCertPinning = true,
            debugMode = true
        )
    }
}

File: demo-app/ios/ObserveSDK/Core/ObserveConfig.swift
//
//  ObserveConfig.swift
//  ObserveSDK
//
//  Configuration for Observe SDK
//

import Foundation

/// Configuration for ObserveSDK
public struct ObserveConfig {
    
    /// Enable/disable SDK
    public let enabled: Bool
    
    /// Auto-start observation on initialization
    public let autoStart: Bool
    
    /// App version string
    public let appVersion: String
    
    /// Server URL for uploading events (optional)
    public let serverUrl: String?
    
    /// Number of events to buffer before export
    public let eventBufferSize: Int
    
    /// Maximum number of stored event files
    public let maxStoredFiles: Int
    
    /// Interval for flushing events (milliseconds)
    public let flushIntervalMs: Int64
    
    /// Enable network request/response capture
    public let enableNetworkCapture: Bool
    
    /// Enable UI hierarchy capture
    public let enableHierarchyCapture: Bool
    
    /// Performance mode (reduces event detail)
    public let performanceMode: Bool
    
    // MARK: - Initializer
    
    public init(
        enabled: Bool = true,
        autoStart: Bool = true,
        appVersion: String = "1.0.0",
        serverUrl: String? = nil,
        eventBufferSize: Int = 500,
        maxStoredFiles: Int = 10,
        flushIntervalMs: Int64 = 30000,
        enableNetworkCapture: Bool = true,
        enableHierarchyCapture: Bool = true,
        performanceMode: Bool = false
    ) {
        self.enabled = enabled
        self.autoStart = autoStart
        self.appVersion = appVersion
        self.serverUrl = serverUrl
        self.eventBufferSize = eventBufferSize
        self.maxStoredFiles = maxStoredFiles
        self.flushIntervalMs = flushIntervalMs
        self.enableNetworkCapture = enableNetworkCapture
        self.enableHierarchyCapture = enableHierarchyCapture
        self.performanceMode = performanceMode
    }
    
    // MARK: - Presets
    
    /// Development configuration (high detail, frequent exports)
    public static func development(appVersion: String = "dev") -> ObserveConfig {
        return ObserveConfig(
            enabled: true,
            autoStart: true,
            appVersion: appVersion,
            serverUrl: nil,
            eventBufferSize: 100,
            maxStoredFiles: 20,
            flushIntervalMs: 10000,
            enableNetworkCapture: true,
            enableHierarchyCapture: true,
            performanceMode: false
        )
    }
    
    /// Production configuration (SDK disabled)
    public static func production() -> ObserveConfig {
        return ObserveConfig(
            enabled: false,
            autoStart: false,
            appVersion: "prod",
            serverUrl: nil,
            eventBufferSize: 0,
            maxStoredFiles: 0,
            flushIntervalMs: 0,
            enableNetworkCapture: false,
            enableHierarchyCapture: false,
            performanceMode: false
        )
    }
    
    /// Test configuration (SDK disabled, for automation)
    public static func test() -> ObserveConfig {
        return ObserveConfig(
            enabled: false,
            autoStart: false,
            appVersion: "test",
            serverUrl: nil,
            eventBufferSize: 0,
            maxStoredFiles: 0,
            flushIntervalMs: 0,
            enableNetworkCapture: false,
            enableHierarchyCapture: false,
            performanceMode: false
        )
    }
}

